server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -85,14 +86,17 @@ public QueryResource(
       QuerySegmentWalker texasRanger,
       ServiceEmitter emitter,
       RequestLogger requestLogger,
-      QueryManager queryManager
+      QueryManager queryManager,
+      InjectableValues guiceInjectableValues
   )
   {
     this.config = config;
     this.jsonMapper = jsonMapper.copy();
+    this.jsonMapper.setInjectableValues(guiceInjectableValues); // Workaround for https://github.com/FasterXML/jackson-databind/issues/696
```

Comment:
That could break if the current behavior relies on something *other than* ObjectMapper to do the closing. I'll look at it more when I have fresh eyes.